src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -44,13 +45,12 @@ public BitArray(int length, bool defaultValue)
                 throw new ArgumentOutOfRangeException(nameof(length), length, SR.ArgumentOutOfRange_NeedNonNegNum);
             }
 
-            m_array = new int[GetArrayLength(length, BitsPerInt32)];
+            m_array = new int[GetInt32ArrayLengthFromBitLength(length)];
             m_length = length;
 
-            int fillValue = defaultValue ? unchecked(((int)0xffffffff)) : 0;
-            for (int i = 0; i < m_array.Length; i++)
+            if (defaultValue)
```

Comment:
Just to confirm, is it fine to skip setting the array values to 0 after declaring it? Will the array always be zero-initialized (or `default-value` initialized)? cc @jkotas

Reply:
Yes, arrays are always initialized with zeros.